torchgeo/datasets/mapinwild.py

```diff
@@ -0,0 +1,479 @@
+            split: one of "train", "validation", or "test"
+            transforms: a function/transform that takes input sample and its target as
+                entry and returns a transformed version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            AssertionError: if ``split`` argument is invalid
+            ImportError: if pandas is not installed
+        """
+        assert split in ["train", "validation", "test"]
+
+        self.checksum = checksum
+        self.root = root
+        self.transforms = transforms
+        self.modality = modality
+        self.download = download
+
+        try:
+            import pandas as pd  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "pandas is not installed and is required to use this dataset"
+            )
+
+        modality.append("split_ids")
+
+        for modal in modality:
+            for modality_link in self.modality_urls[modal]:
+                os.path.join(self.url, modality_link)
+                if modality_link.endswith("csv"):
+                    self._verify(url=modality_link, md5=None)
+                else:
+                    self._verify(
+                        url=modality_link,
+                        md5=self.md5s[os.path.split(modality_link)[1]],
+                    )
+
+            #  Merge modalities downloaded in two parts
+            if (
+                download
+                and modal not in os.listdir(self.root)
+                and len(self.modality_urls[modal]) == 2
+            ):
+                self._merge_parts(modal)
+
+        # Masks will be loaded seperately in the :meth:`__getitem__`
+        if "mask" in self.modality:
+            self.modality.remove("mask")
+
+        # Split IDs has been downloaded and not needed in the list
+        if "split_ids" in self.modality:
+            self.modality.remove("split_ids")
+
+        if os.path.exists(os.path.join(self.root, "split_IDs.csv")):
+            split_dataframe = pd.read_csv(os.path.join(self.root, "split_IDs.csv"))
+            self.ids = split_dataframe[split].dropna().values.tolist()
+            self.ids = [int(i) for i in self.ids]
+
+    def __getitem__(self, index: int) -> dict[str, Tensor]:
+        """Return an index within the dataset.
+
+        Args:
+            index: index to return
+
+        Returns:
+            data and label at that index
+        """
+        list_modals = []
+        id = self.ids[index]
+
+        mask = self._load_raster(id, "mask")
+        mask[mask != 0] = 1
+
+        for mode in self.modality:
+            mode = mode.upper() if mode in ["esa_wc", "viirs"] else mode
+            data = self._load_raster(id, mode)
+            list_modals.append(data)
+
+        image = torch.cat(list_modals, dim=0)
+
+        if self.transforms is not None:
```

Comment:
Transforms should be applied _after_ rearranging the dimensions. All of our built-in transforms assume the data is already in the right shape. You can also get rid of the sample_trans and transformed variables and just have a single sample variable that gets updated.